db/db.py

```diff
@@ -271,27 +281,78 @@ class Job(Base):
     timeout_seconds = Column(Integer, nullable=False)
     email_notifications = Column(JSON, nullable=False)
     workspace_id = Column(String, ForeignKey("workspaces.id"), nullable=False)
-    workspace = relationship(Workspace)
     new_cluster = Column(JSON)
+    existing_cluster_id = Column(String)
     schedule_quartz_cron_expression = Column(String)
     schedule_timezone_id = Column(String)
     task_type = Column(String)
-    notebook_path = Column(String)
-    notebook_revision_timestamp = Column(Integer)
+    task_parameters = Column(JSON)
     max_concurrent_runs = Column(Integer)
 
+    workspace = relationship(Workspace)
+    jobruns = relationship("JobRun",
+                           back_populates='job',
+                           order_by="JobRun.start_time")
+
+    def runs(self, last=None, since_days=None):
+        runs = self.jobruns
+
+        if since_days is not None:
+            runs = [run for run in runs if run.start_time >= since(since_days)]
+
+        if last is not None:
+            runs = runs[-last:]
+
+        return runs
+
+    def num_runs(self, last=None, since_days=None):
+        return len(self.runs(last, since_days))
+
+    def duration(self, last=None, since_days=None):
+        return sum([run.duration() for run in self.runs(last, since_days)])
+
+    def dbu(self, last=None, since_days=None):
+        return sum([run.dbu() for run in self.runs(last, since_days)])
+
+    def cost(self, price_config, last=None, since_days=None):
+        return sum([run.cost(price_config)
+                    for run in self.runs(last, since_days)])
+
+    def run_df(self, price_config=None, last=None, since_days=None):
+        return pd.DataFrame([run.to_dict(price_config)
+                             for run in self.runs(last, since_days)])
+
+    def run_stats(self, aggs, price_config=None, last=None, since_days=None):
```

Comment:
`run_stats` seems to be a bit strange naming. `calculate` maybe? I'm not sure if it's better.

Reply:
Agree, I'll rename it.

Reply:
On second thought, it is really out of place, removed it entirely. Code moved to required place.